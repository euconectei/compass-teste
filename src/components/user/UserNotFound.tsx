type UserNotFoundType = {
  username: string;
};

const UserNotFound = ({ username }: UserNotFoundType) => {
  return (
    <div className='alert alert-info'>
      O usuário {username} não foi encontrado.
    </div>
  );
};

export default UserNotFound;
