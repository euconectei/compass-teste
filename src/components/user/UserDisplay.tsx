import { UserType } from '../../types/UserType';
import UserNotFound from './UserNotFound';

type UserDisplayComponentType = {
  user: UserType;
  username: string;
};

const UserDisplay = ({ user, username }: UserDisplayComponentType) => {
  if (Object.keys(user).length === 0) {
    return <UserNotFound username={username} />;
  }

  return (
    <>
      <div className='m-3'>
        <div className='card user-card'>
          <img
            src={user.avatar_url}
            alt={`Foto de ${user.login}`}
            className='card-img-top'
          />
          <div className='card-body'>
            <h5 className='card-title'>{user.login}</h5>
            <ul className='list-group list-group-flush'>
              <li className='list-group-item'>
                <span className='font-weight-bold'>Nome:</span> {user.name}
              </li>
              <li className='list-group-item'>
                <span className='font-weight-bold'>E-mail:</span> {user.email}
              </li>
              <li className='list-group-item'>
                <span className='font-weight-bold'>Reposit√≥rios:</span>{' '}
                {user.public_repos}
              </li>
              <li className='list-group-item'>
                <span className='font-weight-bold'>Seguidores:</span>{' '}
                {user.followers}
              </li>
              <li className='list-group-item'>
                <span className='font-weight-bold'>Seguindo:</span>{' '}
                {user.following}
              </li>
            </ul>
            <hr className='divider' />
            <div className='d-flex justify-content-between'>
              <a href={`/${user.login}/repos`} className='btn btn-primary'>
                Repos
              </a>
              <a href={`/${user.login}/starred`} className='btn btn-primary'>
                Starred
              </a>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default UserDisplay;
