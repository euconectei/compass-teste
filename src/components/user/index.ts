import UserDisplay from './UserDisplay';
import UserNotFound from './UserNotFound';

export { UserDisplay, UserNotFound };
