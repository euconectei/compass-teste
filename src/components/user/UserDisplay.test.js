import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { userMock } from '../../types/__mocks__/user.mock';
import UserDisplay from './UserDisplay';

describe('Testing UserDisplay Component', () => {
  it('shold render the component', () => {
    const component = renderer.create(<UserDisplay user={userMock} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with no user', () => {
    render(<UserDisplay user={{}} username='aaawewss' />);
    expect(
      screen.getByText('O usuário aaawewss não foi encontrado.')
    ).toBeInTheDocument();
  });
});
