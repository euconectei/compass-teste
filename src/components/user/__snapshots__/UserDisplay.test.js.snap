// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UserDisplay Component shold render the component 1`] = `
<div
  className="m-3"
>
  <div
    className="card user-card"
  >
    <img
      alt="Foto de euconectei"
      className="card-img-top"
      src="https://avatars.githubusercontent.com/u/1842710?v=4"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        euconectei
      </h5>
      <ul
        className="list-group list-group-flush"
      >
        <li
          className="list-group-item"
        >
          <span
            className="font-weight-bold"
          >
            Nome:
          </span>
           
          Rafael França
        </li>
        <li
          className="list-group-item"
        >
          <span
            className="font-weight-bold"
          >
            E-mail:
          </span>
           
        </li>
        <li
          className="list-group-item"
        >
          <span
            className="font-weight-bold"
          >
            Repositórios:
          </span>
           
          40
        </li>
        <li
          className="list-group-item"
        >
          <span
            className="font-weight-bold"
          >
            Seguidores:
          </span>
           
          35
        </li>
        <li
          className="list-group-item"
        >
          <span
            className="font-weight-bold"
          >
            Seguindo:
          </span>
           
          7
        </li>
      </ul>
      <hr
        className="divider"
      />
      <div
        className="d-flex justify-content-between"
      >
        <a
          className="btn btn-primary"
          href="/euconectei/repos"
        >
          Repos
        </a>
        <a
          className="btn btn-primary"
          href="/euconectei/starred"
        >
          Starred
        </a>
      </div>
    </div>
  </div>
</div>
`;
